File: backend/src/main/java/com/nexus/backend/service/MatchingService.java
package com.nexus.backend.service;

import com.nexus.backend.dto.response.GeminiRecommendationResponse;
import com.nexus.backend.dto.response.MatchResponse;

import java.util.List;

public interface MatchingService {

    // =========================================
    // Get Smart Project Matches
    // Java-based skill matching
    // =========================================

    List<MatchResponse> getProjectMatches();

    // =========================================
    // Generate AI Recommendation
    // for one selected project
    // =========================================

    GeminiRecommendationResponse
    generateProjectRecommendation(Long projectId);
}

File: frontend/src/services/matchingService.js
import api from "./api";

// ==========================================
// GET SMART PROJECT MATCHES
//
// Java smart matching only.
// NO GEMINI CALL.
// ==========================================

export const getProjectMatches = async () => {
  const response = await api.get("/matching/projects");

  return response.data;
};


// ==========================================
// GET MATCH HISTORY
// ==========================================

export const getMatchHistory = async () => {
  const response = await api.get("/match-history");

  return response.data;
};


// ==========================================
// GENERATE AI RECOMMENDATION
//
// Gemini is called only when the user
// clicks "Generate AI Analysis".
// ==========================================

export const generateAIRecommendation = async (
    projectId
) => {

  const response =
      await api.post(
          `/matching/projects/${projectId}/ai-recommendation`
      );

  return response.data;
};